@market/modules/Seaport/SeaportFillOrder.tsx

```diff
@@ -0,0 +1,132 @@
+import { Button } from 'components/Button'
+import { OpenSeaIcon } from 'components/Icon'
+import { mediumFont } from 'styles/styles.css'
+
+import React, { Dispatch, SetStateAction, useEffect, useMemo } from 'react'
+
+import { MarketModalHeading } from '@market/components'
+import { useValidateSeaportContractCall } from '@market/modules/Seaport/hooks/useValidateSeaportContractCall'
+import { useModal } from '@modal'
+import { formatContractError, useAuth } from '@shared'
+import { PrintError } from '@shared/components/PrintError'
+import { Flex, Heading, Paragraph, Separator, Stack } from '@zoralabs/zord'
+
+import { CommonSeaportFillOrderProps } from './SeaportFillOrderFlow'
+import { useSeaportTransaction } from './hooks/useSeaportTransaction'
+
+interface SeaportFillOrderProps extends CommonSeaportFillOrderProps {
+  setIsFilled: Dispatch<SetStateAction<boolean>>
+}
+
+export function SeaportFillOrder({
+  order,
+  nft,
+  userAddress,
+  setIsFilled,
+  ...props
+}: SeaportFillOrderProps) {
+  const { requestClose } = useModal()
+
+  const { offchainOrder } = order
+  const { calldata, contractAddress, price } = offchainOrder
+  const { chainTokenPrice } = price
+  const ethPrice = useMemo(() => chainTokenPrice.decimal, [chainTokenPrice])
+  const { balance: walletBalance } = useAuth()
+
+  console.log('chainTokenPrice', chainTokenPrice)
+  console.log('walletBalance', walletBalance)
+
+  const contractCall = useMemo(() => {
+    return {
+      caller_address: userAddress!, // user address ||||| TODO: okay, this is stupid. But the SeaPort validation requires that the user has sufficient funds. If we want to validate regardless, maybe we just use some whala's address for confirmation (eg. the Nouns DAO treasury address)
```

Comment:
I thought you use Nouns treasury address here

Reply:
@olegakbarov for filling the order, we check against the user address. For displaying all of the orders via SSR, we use the treasury address